src/color.h

```diff
@@ -15,161 +170,35 @@ HI = 1,
- c_blue_white  = COLOR_PAIR(37),
- c_ltblue_white  = COLOR_PAIR(37) | A_BOLD,
- c_magenta_white  = COLOR_PAIR(38),
- c_pink_white  = COLOR_PAIR(38) | A_BOLD,
- c_cyan_white  = COLOR_PAIR(39),
- c_ltcyan_white  = COLOR_PAIR(39) | A_BOLD,
- c_black_green  = COLOR_PAIR(40),
- c_dkgray_green  = COLOR_PAIR(40) | A_BOLD,
- c_ltgray_green  = COLOR_PAIR(41),
- c_white_green  = COLOR_PAIR(41) | A_BOLD,
- c_red_green  = COLOR_PAIR(42),
- c_ltred_green  = COLOR_PAIR(42) | A_BOLD,
- c_green_green  = COLOR_PAIR(43),
- c_ltgreen_green  = COLOR_PAIR(43) | A_BOLD,
- c_brown_green  = COLOR_PAIR(44),
- c_yellow_green  = COLOR_PAIR(44) | A_BOLD,
- c_blue_green  = COLOR_PAIR(45),
- c_ltblue_green  = COLOR_PAIR(45) | A_BOLD,
- c_magenta_green  = COLOR_PAIR(46),
- c_pink_green  = COLOR_PAIR(46) | A_BOLD,
- c_cyan_green  = COLOR_PAIR(47),
- c_ltcyan_green  = COLOR_PAIR(47) | A_BOLD,
- c_black_yellow  = COLOR_PAIR(48),
- c_dkgray_yellow  = COLOR_PAIR(48) | A_BOLD,
- c_ltgray_yellow  = COLOR_PAIR(49),
- c_white_yellow  = COLOR_PAIR(49) | A_BOLD,
- c_red_yellow  = COLOR_PAIR(50),
- c_ltred_yellow  = COLOR_PAIR(50) | A_BOLD,
- c_green_yellow  = COLOR_PAIR(51),
- c_ltgreen_yellow  = COLOR_PAIR(51) | A_BOLD,
- c_brown_yellow  = COLOR_PAIR(52),
- c_yellow_yellow  = COLOR_PAIR(52) | A_BOLD,
- c_blue_yellow  = COLOR_PAIR(53),
- c_ltblue_yellow  = COLOR_PAIR(53) | A_BOLD,
- c_magenta_yellow  = COLOR_PAIR(54),
- c_pink_yellow  = COLOR_PAIR(54) | A_BOLD,
- c_cyan_yellow  = COLOR_PAIR(55),
- c_ltcyan_yellow  = COLOR_PAIR(55) | A_BOLD,
- c_black_magenta  = COLOR_PAIR(56),
- c_dkgray_magenta  = COLOR_PAIR(56) | A_BOLD,
- c_ltgray_magenta  = COLOR_PAIR(57),
- c_white_magenta  = COLOR_PAIR(57) | A_BOLD,
- c_red_magenta  = COLOR_PAIR(58),
- c_ltred_magenta  = COLOR_PAIR(58) | A_BOLD,
- c_green_magenta  = COLOR_PAIR(59),
- c_ltgreen_magenta  = COLOR_PAIR(59) | A_BOLD,
- c_brown_magenta  = COLOR_PAIR(60),
- c_yellow_magenta  = COLOR_PAIR(60) | A_BOLD,
- c_blue_magenta  = COLOR_PAIR(61),
- c_ltblue_magenta  = COLOR_PAIR(61) | A_BOLD,
- c_magenta_magenta  = COLOR_PAIR(62),
- c_pink_magenta  = COLOR_PAIR(62) | A_BOLD,
- c_cyan_magenta  = COLOR_PAIR(63),
- c_ltcyan_magenta  = COLOR_PAIR(63) | A_BOLD,
- c_black_cyan  = COLOR_PAIR(64),
- c_dkgray_cyan  = COLOR_PAIR(64) | A_BOLD,
- c_ltgray_cyan  = COLOR_PAIR(65),
- c_white_cyan  = COLOR_PAIR(65) | A_BOLD,
- c_red_cyan  = COLOR_PAIR(66),
- c_ltred_cyan  = COLOR_PAIR(66) | A_BOLD,
- c_green_cyan  = COLOR_PAIR(67),
- c_ltgreen_cyan  = COLOR_PAIR(67) | A_BOLD,
- c_brown_cyan  = COLOR_PAIR(68),
- c_yellow_cyan  = COLOR_PAIR(68) | A_BOLD,
- c_blue_cyan  = COLOR_PAIR(69),
- c_ltblue_cyan  = COLOR_PAIR(69) | A_BOLD,
- c_magenta_cyan  = COLOR_PAIR(70),
- c_pink_cyan  = COLOR_PAIR(70) | A_BOLD,
- c_cyan_cyan  = COLOR_PAIR(71),
- c_ltcyan_cyan  = COLOR_PAIR(71) | A_BOLD
+typedef int nc_color;
 
+class clColors {
+    public:
+        clColors() {};
+
+        nc_color get(const std::string &sName);
+        std::string get(const nc_color &color);
```

Comment:
Taking a const reference is only useful if the object is big or copying it is expensive. Neither is true for an int (which nc_color is). I recommend therefor to take the nc_color parameters as by-value, through a const reference is not incorrect.